truss/contexts/image_builder/cache_warmer.py

```diff
@@ -1,25 +1,41 @@
 import sys
 from pathlib import Path
 
+from google.cloud import storage
 from huggingface_hub import hf_hub_download
 
 
-def download_file(repo_name, file_name, revision_name=None):
-    secret = None
-    secret_path = Path("/etc/secrets/hf_access_token")
-
-    if secret_path.exists():
-        secret = secret_path.read_text().strip()
-    try:
-        hf_hub_download(repo_name, file_name, revision=revision_name, token=secret)
-    except FileNotFoundError:
-        raise RuntimeError(
-            "Hugging Face repository not found (and no valid secret found for possibly private repository)."
-        )
+def download_file(
+    repo_name, file_name, revision_name=None, key_file="/app/data/service_account.json"
+):
+    # Check if repo_name starts with "gs://"
+    if "gs://" in repo_name:
+        # Create directory if not exist
+        repo_name = repo_name.replace("gs://", "")
+        cache_dir = Path(f"/app/hf_cache/{repo_name}")
+        cache_dir.mkdir(parents=True, exist_ok=True)
+
+        # Connect to GCS storage
+        try:
+            storage_client = storage.Client.from_service_account_json(key_file)
```

Comment:
to optimize later: would be great if we only make this client once and re-use it for all the file downloads.